src/main/java/kr/allcll/backend/support/schedule/ScheduleStorage.java

```diff
@@ -11,10 +13,24 @@
 public class ScheduleStorage {
 
     private final Map<String, ScheduledFuture<?>> scheduledPinTasks = new ConcurrentHashMap<>();
-    private ScheduledFuture<?> nonMajorSchedule;
+    private Optional<ScheduledFuture<?>> nonMajorSchedule;
+
+    public ScheduleStorage() {
+        this.nonMajorSchedule = Optional.empty();
+    }
 
     public void cancelNonMajorSchedule() {
-        nonMajorSchedule.cancel(true);
+        nonMajorSchedule.ifPresent(scheduledFuture -> scheduledFuture.cancel(true));
+    }
+
+    public boolean isNonMajorScheduleRunning() {
+        return nonMajorSchedule
+            .map(scheduledFuture -> scheduledFuture.state().equals(State.RUNNING))
```

Comment:
```suggestion
            .map(scheduledFuture -> State.RUNNING.equals(scheduledFuture.state()))
```

항상 equals는 상수가 앞에 나와야 한답니다 😀

Reply:
아... 옵션 엔터로 인텔리제이가 자동으로 만들어준걸 너무 믿었네요 신경쓸게요